### ISSUE TITLE
SPARSE_NORMAL_CHOLESKY performance has regressed 

### ISSUE THREAD

**Issue by sandwichmaker** [ceres-solver/ceres-solver#905]
bundle_adjuster at 1.14.10 with 

```
--input problem-1778-993923-pre.txt -linear_solver sparse_normal_cholesky -ordering auto -num_threads 1

CHOLMOD version 3.0.14, Oct 22, 2019: Symbolic Analysis: status: OK
  Architecture: Mac
    sizeof(int):      4
    sizeof(SuiteSparse_long):  8
    sizeof(void *):   8
    sizeof(double):   8
    sizeof(Int):      4 (CHOLMOD's basic integer)
    sizeof(BLAS_INT): 4 (integer used in the BLAS)
  Results from most recent analysis:
    Cholesky flop count: 1.3218e+11
    Nonzeros in L:       1.7817e+08
  memory blocks in use:          11
  memory in use (MB):         309.5
  peak memory usage (MB):     905.1
  maxrank:    update/downdate rank:   8
  supernodal control: 1 40 (supernodal if flops/lnz >= 40)
  nmethods:   number of ordering methods to try: 1
    method 0: natural
        flop count: 1.3218e+11
        nnz(L):     1.7817e+08
  OK
I20221030 16:37:48.307452 4331665 wall_time.cc:79] 

SparseNormalCholeskySolver::Solve
                                   Delta   Cumulative
                     Compute RHS :    0.09363      0.09363
                        Diagonal :    0.02564      0.11927
                          Append :    0.01949      0.13876
                     Append Rows :    0.01609      0.15485
   InnerProductComputer::Compute :    1.06970      1.22456
                  Factor & Solve :   24.23193     25.45648
                           Total :    0.00000     25.45648

Time (in seconds):
Preprocessor                        13.763938

  Residual only evaluation           1.253244 (5)
  Jacobian & residual evaluation     6.719860 (6)
  Linear solver                    137.836717 (5)
Minimizer                          148.104302

Postprocessor                        0.062778
Total                              161.931018
```

So roughly 25 seconds per linear solve.

bundle_adjuster at HEAD

```
--input problem-1778-993923-pre.txt -linear_solver sparse_normal_cholesky -linear_solver_ordering auto -num_threads 1 

CHOLMOD version 3.0.14, Oct 22, 2019: Symbolic Analysis: status: OK
  Architecture: Mac
    sizeof(int):      4
    sizeof(SuiteSparse_long):  8
    sizeof(void *):   8
    sizeof(double):   8
    sizeof(Int):      4 (CHOLMOD's basic integer)
    sizeof(BLAS_INT): 4 (integer used in the BLAS)
  Results from most recent analysis:
    Cholesky flop count: 1.3218e+11
    Nonzeros in L:       1.7817e+08
  memory blocks in use:          11
  memory in use (MB):         309.4
  peak memory usage (MB):    1454.7
  maxrank:    update/downdate rank:   8
  supernodal control: 1 40 (supernodal if flops/lnz >= 40)
  nmethods:   number of ordering methods to try: 1
    method 0: natural
        flop count: 1.3218e+11
        nnz(L):     1.7817e+08
  OK

SparseNormalCholeskySolver::Solve
                                        Delta   Cumulative
                     Compute RHS :    0.07391      0.07391
                        Diagonal :    0.02629      0.10020
                          Append :    0.03533      0.13554
                     Append Rows :    0.01605      0.15159
   InnerProductComputer::Compute :    0.83975      0.99134
  SparseCholesky::FactorAndSolve :   31.28317     32.27451
                           Total :    0.00002     32.27453

Time (in seconds):
Preprocessor                        13.699032

  Residual only evaluation           1.778521 (5)
  Jacobian & residual evaluation     6.620192 (6)
  Linear solver                    176.330795 (5)
Minimizer                          187.974501

Postprocessor                        0.044577
Total                              201.718110
```

**Comment by sandwichmaker** [comment]
On problem-356-226730-pre.txt

1.14
Linear solver                     49.294211 (5)

HEAD
Linear solver                     54.416987 (5)

**Comment by sandwichmaker** [comment]
2.0.0 seems to be fine, and looks like this has to do with an increase in peak memory usage.

**Comment by sandwichmaker** [comment]
The offending commit seems to be d09f7e9d, where I enabled post ordering for all orderings, and this seems to increase peak memory use as well as factorization time.

**Comment by sandwichmaker** [comment]
Fixed.

**Comment by rphel** [comment]
Hello! Is there any plan to release 2.2 anytime soon?
ceres 2.1 is the only version supported by [conda-forge](https://anaconda.org/conda-forge/ceres-solver/files) latest builds and is of course affected by this issue. It would be really great to get this fix in a 2.2 or a 2.1.1 version. If you need an extra set of hands, just let me know and thank you for your amazing work!

**Comment by sandwichmaker** [comment]
@rphel we are hoping for a new release later this year once the gpu bundle adjustment stuff lands which is a couple of patches out.
